### PR TITLE
fixes updb enrollment, underlying error types changed

### PR DESCRIPTION
**Pull request by andrewpmartinez**
Instead of checking for specific error types, try blindly to ensure success.

### CODE REVIEW

File: ziti/enroll/enroll.go [hunk 1]
```diff
@@ -139,26 +139,20 @@ func EnrollUpdb(enFlags EnrollmentFlags) error {
...
+		rootCaPool.AddCert(enFlags.Token.SignatureCert)
 
-		return fmt.Errorf("unhandled error: %v", err)
+		for _, xcert := range FetchCertificates(ztApiRoot, rootCaPool) {
+			allowedCerts = append(allowedCerts, xcert)
+			caPool.AddCert(xcert)
+		}
+		//try again
+		if err := enrollUpdb(enFlags.Username, enFlags.Password, enFlags.Token, caPool); err != nil {
+			return fmt.Errorf("unabled to enroll after fetching server certs: %v", err)
```
Comment (dovholuknf) [inline ziti/enroll/enroll.go:152]: i hate being 'unabled'! :) (typo)

File: ziti/enroll/enroll.go [hunk 2]
```diff
@@ -139,26 +139,20 @@ func EnrollUpdb(enFlags EnrollmentFlags) error {
...
+		pfxlog.Logger().Debug("fetching certificates from server")
+		rootCaPool := x509.NewCertPool()
+		rootCaPool.AddCert(enFlags.Token.SignatureCert)
 
-		return fmt.Errorf("unhandled error: %v", err)
+		for _, xcert := range FetchCertificates(ztApiRoot, rootCaPool) {
+			allowedCerts = append(allowedCerts, xcert)
+			caPool.AddCert(xcert)
+		}
+		//try again
```
Comment (dovholuknf) [inline ziti/enroll/enroll.go:150]: this is not "trying again" now, it's trying for the first time

File: ziti/enroll/enroll.go
Comment (dovholuknf) [inline ziti/enroll/enroll.go:1]: Do we have any doc that states explicitly you will now need to include the entire chain in the cas store if you're using a third party 'verifiable' CA?  Before, it'd use the OS trust store and now it exclusively fetches the certificates every time, which is fine, but we need to make sure people know that they need to add the full chain of trust for this to work now, assuming I'm understanding this change correcctly

Comment (andrewpmartinez) [reply ziti/enroll/enroll.go:1]: This only changes the error detection. Prior to this, the code checked for specific error types, which have changed with golang upgrades. This change removes the type assertions and looks for any error.

Other than that, I do not believe any behavior has changed.

**Review by dovholuknf — APPROVED** [review]

**Pull request merged**